pkg/client/openstack/admin/client.go

```diff
@@ -251,3 +260,54 @@ func (c *adminClient) getRoleID(roleName string) (string, error) {
 	return "", fmt.Errorf("Role %s not found", roleName)
 
 }
+
+func (c *adminClient) CreateContainer(containerName, serviceUserName, serviceUserDomainName, serviceUserProjectID string) error {
+	listOpts := endpoints.ListOpts{
+		Availability: "public",
+	}
+
+	allPages, err := endpoints.List(c.IdentityClient, listOpts).AllPages()
+	if err != nil {
+		return err
+	}
+
+	allEndpoints, err := endpoints.ExtractEndpoints(allPages)
+	if err != nil {
+		return err
+	}
+
+	var swiftEndpoint endpoints.Endpoint
+	for _, endpoint := range allEndpoints {
+		if endpoint.Name == "swift" {
+			swiftEndpoint = endpoint
+			break
+		}
+	}
+	endpointURL := strings.Replace(swiftEndpoint.URL, "%(tenant_id)s", serviceUserProjectID, 1)
+
+	storageClient, err := openstack.NewObjectStorageV1(c.ProviderClient, gophercloud.EndpointOpts{})
+	if err != nil {
+		return err
+	}
+	storageClient.Endpoint = endpointURL
+
+	domainID, err := c.getDomainID(serviceUserDomainName)
+	if err != nil {
+		return err
+	}
+
+	serviceUser, err := c.getUserByName(serviceUserName, domainID)
+	if err != nil {
+		return err
+	}
+
+	acl := fmt.Sprintf("%s:%s", serviceUserProjectID, serviceUser.ID)
+	createOpts := containers.CreateOpts{
+		ContainerRead:  acl,
+		ContainerWrite: acl,
+	}
+
+	res := containers.Create(storageClient, containerName, createOpts)
```

Comment:
I believe this should be like this instead:
```suggestion
	_, err == containers.Create(storageClient, containerName, createOpts).Extract()
```